src/test/compile-fail/const-pattern-irrefutable.rs

```diff
@@ -13,17 +13,17 @@ mod foo {
     pub const d: u8 = 2;
 }
 
-use foo::b as c; //~ NOTE is imported here
-use foo::d; //~ NOTE is imported here
+use foo::b as c;
+use foo::d;
 
-const a: u8 = 2; //~ NOTE is defined here
+const a: u8 = 2;
 
 fn main() {
-    let a = 4; //~ ERROR let bindings cannot shadow constants
-               //~^ NOTE cannot be named the same as a constant
-    let c = 4; //~ ERROR let bindings cannot shadow constants
-               //~^ NOTE cannot be named the same as a constant
-    let d = 4; //~ ERROR let bindings cannot shadow constants
-               //~^ NOTE cannot be named the same as a constant
+    let a = 4; //~ ERROR refutable pattern in local binding: `_` not covered
```

Comment:
Can we provide some better hint as to why this error is occurring? I could see this being rather perplexing.